components/lib/chip/Chip.spec.js

```diff
@@ -0,0 +1,44 @@
+import { fireEvent, render, within } from '@testing-library/react';
+import { Chip } from './Chip';
+
+import '@testing-library/jest-dom';
+
+describe('Chip', () => {
+    test('when removable is true it returns with remove icon', () => {
+        const removeOn = jest.fn();
+        const { container } = render(<Chip removable onRemove={removeOn} />);
+        const chipRemoveIcon = container.getElementsByClassName('p-chip-remove-icon')[0];
+
+        fireEvent.click(chipRemoveIcon);
+
+        expect(container.getElementsByClassName('p-chip-remove-icon pi pi-times-circle').length).toBe(0);
+        expect(removeOn).toHaveBeenCalledTimes(1);
```

Comment:
We can decide as a team or @mertsincan can decide but I am big fan of "arrange act assert" so something like this.

```js
        // Arrange
        const removeOn = jest.fn();
        const { container } = render(<Chip removable onRemove={removeOn} />);
        const chipRemoveIcon = container.getElementsByClassName('p-chip-remove-icon')[0];

        // Act
        fireEvent.click(chipRemoveIcon);

        // Assert
        expect(container.getElementsByClassName('p-chip-remove-icon pi pi-times-circle').length).toBe(0);
        expect(removeOn).toHaveBeenCalledTimes(1);
```

Reply:
Hi, Testing library is a legit library. Buğra makes this understandable by using the right spaces. Reviews provide that too. With all due respect, I don't believe this is necessary. I wonder why you think it's necessary.

Reply:
@bahadirsofuoglu it is a very common pattern in unit testing in C#/Java etc.

Here is a great explanation: https://docs.telerik.com/devtools/justmock/basic-usage/arrange-act-assert#:~:text=Benefits%20of%20Using%20Arrange%20Act%20Assert&text=Clarifies%20and%20focuses%20attention%20on,many%20different%20things%20at%20once.